lumicks/pylake/kymo.py

```diff
@@ -91,6 +132,20 @@ def timestamps(self) -> np.ndarray:
     def _plot(self, image, **kwargs):
         import matplotlib.pyplot as plt
 
+        def fetch_custom_property(args, field):
```

Comment:
I would call it something like `extract_custom_property` to make the intent clearer. I took me a while to understand this was modifying the arguments dictionary.

Another possibility is to do a list comprehension with conditional expression.
```py
x_lims, y_lims = (args.pop(f) if f in args else None) for f in ("xlim", "ylim")
```
Not sure we're not hurting readability on this one. Maybe a mix of both could do the job better.

Reply:
That's somewhat terse. If we go with just the list comprehension, maybe I'd add a comment above it. As an alternative, I just pushed a version with a list comprehension, but with the function defined before. I'm good with either so take your pick :)

Reply:
🥇 Oh man, that is great actually. Readable with +2, -16 is too good to resist :D